### DISCUSSION TITLE
Coinbase connect issues

### DISCUSSION THREAD

**Discussion by GlobalXBPay** [reown-com/appkit#3019] [General]
We tried WalletConnect Flutter integration for android and iOS app. The metamask integration works end to end where we are able to redirect user and transfer coins between accounts. However, coinbase integration fails. The user is able to get redirected to coinbase but we think the pairing between the DeFi app and coinbase wallet is not established. On debug , we could see that session.topic is null for coinbase. Also we dont see the pairing logs getting outputed unlike Metamask integration.  Let us know if we can share any more details 

A successful metamask integration debug logs
flutter: pairing sendRequest, ...
flutter: [UrlUtils] openRedirect ...
flutter: Handling Publish Message:, 
flutter: Handling Publish Message: ...
flutter: Pairing _onMessageEvent, Received data: {}
flutter: pairing sendResult, id: 
flutter: [W3MService] set local chain eip155:137
flutter: Connected Wallet Address:
flutter: [BlockChainService] Failed to get request eth_getBalance. Response: {"jsonrpc":"2.0","error":{"code":-32700,"message":"Parse error","data":"Failed to parse the request body as JSON"}}
flutter: , Status code: 400
flutter: [W3MService] loadAccountData
flutter: Handling Publish Message: 
flutter: pairing sendResult, 
flutter: [W3MService] session event SessionEvent()

Coinbase integration debug logs with session topic as empty 
flutter: [W3MService] _onCoinbaseConnectEvent: {}
flutter: [W3MService] set local chain eip155:1
flutter: Connected Wallet Address:
flutter: client : Client()
flutter: [BlockChainService] Failed to get request eth_getBalance. Response: {"jsonrpc":"2.0","error":{"code":-32600,"message":"invalid json request"}}, Status code: 400
flutter: [W3MService] loadAccountData

**Comment by quetool** [comment]
Hello @GlobalXBPay, can you open an issue on [reown_flutter](https://github.com/reown-com/reown_flutter) repo and close this one? Please add any relevant code of your implementation. Thanks!

**Comment by GlobalXBPay** [comment]
Hi, done. https://github.com/reown-com/reown_flutter/discussions/8

**Reply by quetool** [reply]
Yeah you opened a "discussion" and it was supposed to be an "issue", it's ok anyway. Thanks.
